### PR TITLE
added etherscan station

### PR DESCRIPTION
**Pull request by Guillembonet**
Added etherscan API to gas stations for ethereum mainnet

Signed-off-by: Guillem <guillem@mysterium.network>

### CODE REVIEW

File: transfer/gas/etherscan.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	fmt.Println(string(resp))
+
+	var res etherscanGasPriceResponse
+	err = json.Unmarshal(resp, &res)
+	if err != nil {
+		return nil, err
+	}
+
+	if res.Status != "1" {
+		return nil, errors.New("etherscan API request failed")
```
Comment (tomasmik) [inline transfer/gas/etherscan.go:108]: also provide what happened, status code/message or something. WIthout that the error is a black box.

File: transfer/gas/etherscan.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+		return nil, err
+	}
+	defer response.Body.Close()
+
+	resp, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Println(string(resp))
```
Comment (tomasmik) [inline transfer/gas/etherscan.go:99]: leftover

File: transfer/gas/etherscan.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	"github.com/mysteriumnetwork/payments/units"
+	"github.com/rs/zerolog/log"
+)
+
+// DefaultEtherscanEndpointURI the default etherscan api endpoint.
+const DefaultEtherscanEndpointURI = "https://api.etherscan.io/"
+
+// EtherscanStation represents the etherscan api to retrive gas prices.
+type EtherscanStation struct {
+	APIKey      string
```
Comment (tomasmik) [inline transfer/gas/etherscan.go:23]: probably no need to keep these public

**Review by tomasmik — CHANGES_REQUESTED** [review]

**Review by vkuznecovas — COMMENTED** [review]
What he said.

**Comment by Guillembonet** [comment]
Will do another PR together with refactor with this fixed

**Pull request closed without merging**